graylog2-web-interface/src/components/users/PermissionSelector.jsx

```diff
@@ -90,67 +105,67 @@ const PermissionSelector = React.createClass({
   /*
    * onClick actions for single edits
    */
-  _toggleStreamReadPermissions(stream) {
-    this._toggleReadPermissions('streams', Immutable.Set.of(stream.id));
+  _setStreamReadPermissions(stream) {
+    this._setReadPermissions('streams', Immutable.Set.of(stream.id));
   },
 
-  _toggleStreamEditPermissions(stream) {
-    this._toggleEditPermissions('streams', Immutable.Set.of(stream.id));
+  _setStreamEditPermissions(stream) {
+    this._setEditPermissions('streams', Immutable.Set.of(stream.id));
   },
 
-  _toggleDashboardReadPermissions(dashboard) {
-    this._toggleReadPermissions('dashboards', Immutable.Set.of(dashboard.id));
+  _setDashboardReadPermissions(dashboard) {
+    this._setReadPermissions('dashboards', Immutable.Set.of(dashboard.id));
   },
 
-  _toggleDashboardEditPermissions(dashboard) {
-    this._toggleEditPermissions('dashboards', Immutable.Set.of(dashboard.id));
+  _setDashboardEditPermissions(dashboard) {
+    this._setEditPermissions('dashboards', Immutable.Set.of(dashboard.id));
   },
 
   /*
    * onClick actions for bulk edits
    */
 
-  _toggleAllStreamsRead(streamIds) {
-    this._toggleReadPermissions('streams', streamIds);
+  _setAllStreamsRead(streamIds, allUnset) {
+    this._setReadPermissions('streams', streamIds, allUnset);
   },
 
-  _toggleAllStreamsEdit(streamIds) {
-    this._toggleEditPermissions('streams', streamIds);
+  _setAllStreamsEdit(streamIds, allUnset) {
+    this._setEditPermissions('streams', streamIds, allUnset);
   },
 
-  _toggleAllDashboardsRead(dashboardIds) {
-    this._toggleReadPermissions('dashboards', dashboardIds);
+  _setAllDashboardsRead(dashboardIds, allUnset) {
+    this._setReadPermissions('dashboards', dashboardIds, allUnset);
   },
 
-  _toggleAllDashboardsEdit(dashboardIds) {
-    this._toggleEditPermissions('dashboards', dashboardIds);
+  _setAllDashboardsEdit(dashboardIds, allUnset) {
+    this._setEditPermissions('dashboards', dashboardIds, allUnset);
   },
 
-  _toggleReadPermissions(target, idList) {
+  _setReadPermissions(target, idList, allUnset = true) {
     let added = Immutable.Set.of();
     let deleted = Immutable.Set.of();
 
     idList.forEach((id) => {
       const readTarget = `${target}:read:${id}`;
       const editTarget = `${target}:edit:${id}`;
 
-      if (this.props.permissions.contains(readTarget)) {
+      if (this.props.permissions.contains(readTarget) && allUnset) {
```

Comment:
This is a small (and subjective) thing, but in my opinion `shouldClear` or `clearPermissions` would be easier to read than `allUnset`.